medcat/cdb.py

```diff
@@ -376,7 +388,7 @@ def load(cls, path, config=None):
             # Again no idea
             data = dill.load(f)
             if config is None:
-                config = Config.from_dict(data['config'])
+                config = cast(Config, Config.from_dict(data['config']))
```

Comment:
do not really get this one?

Reply:
This is due to the concurrence of inheritance and Optional: `from_dict` is defined by the mixin so it returns an object of type `ConfigMixin` and `config` is of type `Optional[Config]`. So `config = Config.from_dict(data['config'])` would fail the type checking.

Reply:
Is it better to have the config defined as union of Config and ConfigMixing than casting it?

Reply:
The union won't help coz `issubclass(Config, ConfigMixing) == True`. The dilemma here is `Optional[Config]` so you need to use that define the union. Nonetheless, casting is still needed further down the line coz ConfigMixing does not have attributes like `general`.  Removing the config arg or making it compulsory would help.